src/WebJobs.Script.WebHost/Diagnostics/DeferredLogger.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Channels;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics
+{
+    /// <summary>
+    /// A logger that defers log entries to a channel.
+    /// </summary>
+    public class DeferredLogger : ILogger
+    {
+        private readonly Channel<DeferredLogEntry> _channel;
+        private readonly string _categoryName;
+        private readonly IExternalScopeProvider _scopeProvider;
+
+        public DeferredLogger(Channel<DeferredLogEntry> channel, string categoryName, IExternalScopeProvider scopeProvider)
+        {
+            _channel = channel;
+            _categoryName = categoryName;
+            _scopeProvider = scopeProvider;
+        }
+
+        public IDisposable BeginScope<TState>(TState state) => _scopeProvider.Push(state);
+
+        // Restrict logging to errors only
+        public bool IsEnabled(LogLevel logLevel) => logLevel >= LogLevel.Error;
```

Comment:
Did you consider also logging Warnings as well? Might be a good idea.

Reply:
yes, but there are concern about displaying all the initial startup logs in AppInsights. These logs are meant for internal use and could confuse customers. Some of them are at the Warning level. Since the category of these logs is the same as those we want to capture, filtering by category isn’t an option.

We will clean-up all the internal/redundant logs and then allow all the logs to flow through this pipeline. I have been discussing this with Varad and Fabio. This is intentional for now.

Reply:
Are there issues filed tracking the improvements that will unblock this? If not can we create them and ultimately link them here (for context on why we're restricting to errors).

It's important that someone coming back to this code understands why we have this restriction in place.

Reply:
Added link in the comment.